javalib/src/main/scala/java/time/LocalDate.scala

```diff
@@ -0,0 +1,432 @@
+package java.time
+
+import scala.scalajs.js
+
+import java.time.chrono.{IsoEra, Era, IsoChronology, ChronoLocalDate}
+import java.time.temporal._
+
+final class LocalDate private (year: Int, month: Int, dayOfMonth: Int)
+    extends ChronoLocalDate with Serializable {
+  import Preconditions.requireDateTime
+  import ChronoField._
+  import ChronoUnit._
+  import LocalDate._
+
+  requireDateTime(year >= minYear && year <= maxYear,
+      s"Invalid value for year: $year")
+  requireDateTime(month > 0 && month <= 12, s"Invalid value for month: $month")
+
+  private val _month = Month.of(month)
+  private val _isLeapYear = iso.isLeapYear(year)
+
+  requireDateTime(dayOfMonth > 0 && dayOfMonth <= _month.maxLength,
+      s"Invalid value for dayOfMonth: $dayOfMonth")
+  requireDateTime(_isLeapYear || dayOfMonth <= _month.minLength,
+      s"Invalid value for dayOfMonth: $dayOfMonth")
+
+  private val dayOfYear = _month.firstDayOfYear(_isLeapYear) + dayOfMonth - 1
+  private val epochDay = {
+    val year1 = year - 1970
+    val daysBeforeYear = daysBeforeYears(MathJDK8Bridge.floorMod(year1, 400))._2
+    val offset =
+      MathJDK8Bridge.floorDiv(year1, 400).toLong * daysInFourHundredYears
+    offset + daysBeforeYear + dayOfYear - 1
+  }
+  private val dayOfWeek = MathJDK8Bridge.floorMod(epochDay + 3, 7).toInt + 1
```

Comment:
It might be cleaner to put all this fields in the primary constructor and add a secondary constructor (with the current parameters) that checks all the conditions.